### DISCUSSION TITLE
How to type a branded value without cast?

### DISCUSSION THREAD

**Discussion by Pyrolistical** [colinhacks/zod#2956] [Q&A] [answered]
I'm confused on how to type a branded value with casting.

```ts
import { z, BRAND } from 'zod';

const Foo = z.enum(['A', 'B']).brand<'Foo'>();
type Foo = z.infer<typeof Foo>;

// type error Type '"A"' is not assignable to type '("A" | "B") & BRAND<"Foo">'.(2322)
const x: Foo = 'A';

// cast works
const y = 'A' as Foo;

// this silly thing works
const xx: Foo = Object.assign('A', {
  [BRAND]: {
    Foo: true as const,
  },
});
```

Playground: https://stackblitz.com/edit/typescript-otwa6m?file=index.ts

Is there something I'm missing that is simpler? Are we missing a `z.brand('A', 'Foo')`?

**Comment by JacobWeisenburger** [comment]
Is this what you are looking for?

```ts
const foo = z.enum( [ 'A', 'B' ] ).brand<'Foo'>()
type Foo = z.infer<typeof foo>
const x: Foo = foo.parse( 'A' ) // works
```

If you found my answer satisfactory, please consider supporting me. Even a small amount is greatly appreciated. Thanks friend! 🙏
https://github.com/sponsors/JacobWeisenburger

**Reply by Pyrolistical** [reply]
Hmm that does work but is odd. This is using the side effect of a successful parse to brand a value. Whereas I was looking for a more direct "brand this thing". But it does work and avoids the problem of misbranding an invalid value

**Reply by JacobWeisenburger** [reply, marked as answer]
You should probably read this: https://github.com/colinhacks/zod#brand
